src/mods/30-nvidia-safe-installer/install.sh

```
@@ -0,0 +1,196 @@
+        log "Warning: Kernel taints detected after driver install"
+    fi
+    log "Validation successful"
+}
+
+# Rollback to previous snapshot
+rollback() {
+    log "Rolling back NVIDIA driver installation"
+    if mount | grep -q "on / type btrfs"; then
+        # Find latest snapshot and rollback
+        local latest_snapshot
+        latest_snapshot=$(ls -1dt "$SNAPSHOT_DIR"/nvidia-driver-* 2>/dev/null | head -1 || true)
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)